From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Julien Massot <julien.massot@collabora.com>
Date: Fri, 14 Feb 2025 10:05:24 +0100
Subject: configs: fix radxa-nio-12l defconfig

Fix ethernet phy_startup failed by reusing the configuration
from the genio_1200_evk_ufs_defconfig.

After this patch the only difference with the Genio UFS EVK is the
devicetree.

Signed-off-by: Julien Massot <julien.massot@collabora.com>
---
 configs/genio_1200_radxa_nio_12l_d16_defconfig | 13 +++++++++-
 configs/genio_1200_radxa_nio_12l_d4_defconfig  | 13 +++++++++-
 configs/genio_1200_radxa_nio_12l_d8_defconfig  | 13 +++++++++-
 3 files changed, 36 insertions(+), 3 deletions(-)

diff --git a/configs/genio_1200_radxa_nio_12l_d16_defconfig b/configs/genio_1200_radxa_nio_12l_d16_defconfig
index 111111111111..222222222222 100644
--- a/configs/genio_1200_radxa_nio_12l_d16_defconfig
+++ b/configs/genio_1200_radxa_nio_12l_d16_defconfig
@@ -7,7 +7,7 @@ CONFIG_SYS_TEXT_BASE=0x4c000000
 CONFIG_SYS_MALLOC_LEN=0x500000
 CONFIG_SYS_MALLOC_F_LEN=0x4000
 CONFIG_NR_DRAM_BANKS=1
-CONFIG_ENV_SIZE=0x1000
+CONFIG_ENV_SIZE=0x2000
 CONFIG_ENV_OFFSET=0x0
 CONFIG_DM_GPIO=y
 CONFIG_DEFAULT_DEVICE_TREE="genio-1200-radxa-nio-12l-d16"
@@ -20,6 +20,7 @@ CONFIG_DEBUG_UART=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
+CONFIG_LEGACY_IMAGE_FORMAT=y
 # CONFIG_ARCH_FIXUP_FDT_MEMORY is not set
 CONFIG_DEFAULT_FDT_FILE="mediatek/mt8395-radxa-nio-12l"
 # CONFIG_DISPLAY_BOARDINFO is not set
@@ -33,6 +34,7 @@ CONFIG_CMD_BOOTEFI_SELFTEST=y
 CONFIG_CMD_NVEDIT_EFI=y
 CONFIG_CMD_NVEDIT_INFO=y
 # CONFIG_CMD_CRC32 is not set
+CONFIG_CMD_ADC=y
 CONFIG_CMD_CLK=y
 CONFIG_CMD_DFU=y
 CONFIG_CMD_DM=y
@@ -42,6 +44,7 @@ CONFIG_CMD_I2C=y
 # CONFIG_CMD_LOADB is not set
 # CONFIG_CMD_LOADS is not set
 CONFIG_CMD_MBR=y
+CONFIG_CMD_MMC=y
 CONFIG_CMD_UFS=y
 CONFIG_CMD_USB=y
 CONFIG_CMD_USB_MASS_STORAGE=y
@@ -56,6 +59,7 @@ CONFIG_ENV_IMPORT_FDT=y
 CONFIG_REGMAP=y
 CONFIG_SYSCON=y
 CONFIG_DEVRES=y
+CONFIG_ADC_MTK=y
 CONFIG_CLK=y
 CONFIG_DFU_MMC=y
 CONFIG_DFU_UFS=y
@@ -71,6 +75,7 @@ CONFIG_DM_I2C=y
 CONFIG_SYS_I2C_MTK=y
 # CONFIG_INPUT is not set
 # CONFIG_MMC_QUIRKS is not set
+CONFIG_MMC_MTK=y
 CONFIG_PHY_REALTEK=y
 CONFIG_PHY_ETHERNET_ID=y
 CONFIG_DWC_ETH_QOS=y
@@ -80,6 +85,12 @@ CONFIG_PHY_MTK_TPHY=y
 CONFIG_PINCTRL=y
 CONFIG_PINCONF=y
 CONFIG_PINCTRL_MT8195=y
+CONFIG_POWER_DOMAIN=y
+CONFIG_MTK_POWER_DOMAIN=y
+CONFIG_DM_PMIC=y
+CONFIG_MTK_PMIC_WRAP=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_MT6357=y
 CONFIG_DM_RTC=y
 CONFIG_RTC_EMULATION=y
 CONFIG_SCSI=y
diff --git a/configs/genio_1200_radxa_nio_12l_d4_defconfig b/configs/genio_1200_radxa_nio_12l_d4_defconfig
index 111111111111..222222222222 100644
--- a/configs/genio_1200_radxa_nio_12l_d4_defconfig
+++ b/configs/genio_1200_radxa_nio_12l_d4_defconfig
@@ -7,7 +7,7 @@ CONFIG_SYS_TEXT_BASE=0x4c000000
 CONFIG_SYS_MALLOC_LEN=0x500000
 CONFIG_SYS_MALLOC_F_LEN=0x4000
 CONFIG_NR_DRAM_BANKS=1
-CONFIG_ENV_SIZE=0x1000
+CONFIG_ENV_SIZE=0x2000
 CONFIG_ENV_OFFSET=0x0
 CONFIG_DM_GPIO=y
 CONFIG_DEFAULT_DEVICE_TREE="genio-1200-radxa-nio-12l-d4"
@@ -20,6 +20,7 @@ CONFIG_DEBUG_UART=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
+CONFIG_LEGACY_IMAGE_FORMAT=y
 # CONFIG_ARCH_FIXUP_FDT_MEMORY is not set
 CONFIG_DEFAULT_FDT_FILE="mediatek/mt8395-radxa-nio-12l"
 # CONFIG_DISPLAY_BOARDINFO is not set
@@ -33,6 +34,7 @@ CONFIG_CMD_BOOTEFI_SELFTEST=y
 CONFIG_CMD_NVEDIT_EFI=y
 CONFIG_CMD_NVEDIT_INFO=y
 # CONFIG_CMD_CRC32 is not set
+CONFIG_CMD_ADC=y
 CONFIG_CMD_CLK=y
 CONFIG_CMD_DFU=y
 CONFIG_CMD_DM=y
@@ -42,6 +44,7 @@ CONFIG_CMD_I2C=y
 # CONFIG_CMD_LOADB is not set
 # CONFIG_CMD_LOADS is not set
 CONFIG_CMD_MBR=y
+CONFIG_CMD_MMC=y
 CONFIG_CMD_UFS=y
 CONFIG_CMD_USB=y
 CONFIG_CMD_USB_MASS_STORAGE=y
@@ -56,6 +59,7 @@ CONFIG_ENV_IMPORT_FDT=y
 CONFIG_REGMAP=y
 CONFIG_SYSCON=y
 CONFIG_DEVRES=y
+CONFIG_ADC_MTK=y
 CONFIG_CLK=y
 CONFIG_DFU_MMC=y
 CONFIG_DFU_UFS=y
@@ -71,6 +75,7 @@ CONFIG_DM_I2C=y
 CONFIG_SYS_I2C_MTK=y
 # CONFIG_INPUT is not set
 # CONFIG_MMC_QUIRKS is not set
+CONFIG_MMC_MTK=y
 CONFIG_PHY_REALTEK=y
 CONFIG_PHY_ETHERNET_ID=y
 CONFIG_DWC_ETH_QOS=y
@@ -80,6 +85,12 @@ CONFIG_PHY_MTK_TPHY=y
 CONFIG_PINCTRL=y
 CONFIG_PINCONF=y
 CONFIG_PINCTRL_MT8195=y
+CONFIG_POWER_DOMAIN=y
+CONFIG_MTK_POWER_DOMAIN=y
+CONFIG_DM_PMIC=y
+CONFIG_MTK_PMIC_WRAP=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_MT6357=y
 CONFIG_DM_RTC=y
 CONFIG_RTC_EMULATION=y
 CONFIG_SCSI=y
diff --git a/configs/genio_1200_radxa_nio_12l_d8_defconfig b/configs/genio_1200_radxa_nio_12l_d8_defconfig
index 111111111111..222222222222 100644
--- a/configs/genio_1200_radxa_nio_12l_d8_defconfig
+++ b/configs/genio_1200_radxa_nio_12l_d8_defconfig
@@ -7,7 +7,7 @@ CONFIG_SYS_TEXT_BASE=0x4c000000
 CONFIG_SYS_MALLOC_LEN=0x500000
 CONFIG_SYS_MALLOC_F_LEN=0x4000
 CONFIG_NR_DRAM_BANKS=1
-CONFIG_ENV_SIZE=0x1000
+CONFIG_ENV_SIZE=0x2000
 CONFIG_ENV_OFFSET=0x0
 CONFIG_DM_GPIO=y
 CONFIG_DEFAULT_DEVICE_TREE="genio-1200-radxa-nio-12l-d8"
@@ -20,6 +20,7 @@ CONFIG_DEBUG_UART=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
+CONFIG_LEGACY_IMAGE_FORMAT=y
 # CONFIG_ARCH_FIXUP_FDT_MEMORY is not set
 CONFIG_DEFAULT_FDT_FILE="mediatek/mt8395-radxa-nio-12l"
 # CONFIG_DISPLAY_BOARDINFO is not set
@@ -33,6 +34,7 @@ CONFIG_CMD_BOOTEFI_SELFTEST=y
 CONFIG_CMD_NVEDIT_EFI=y
 CONFIG_CMD_NVEDIT_INFO=y
 # CONFIG_CMD_CRC32 is not set
+CONFIG_CMD_ADC=y
 CONFIG_CMD_CLK=y
 CONFIG_CMD_DFU=y
 CONFIG_CMD_DM=y
@@ -42,6 +44,7 @@ CONFIG_CMD_I2C=y
 # CONFIG_CMD_LOADB is not set
 # CONFIG_CMD_LOADS is not set
 CONFIG_CMD_MBR=y
+CONFIG_CMD_MMC=y
 CONFIG_CMD_UFS=y
 CONFIG_CMD_USB=y
 CONFIG_CMD_USB_MASS_STORAGE=y
@@ -56,6 +59,7 @@ CONFIG_ENV_IMPORT_FDT=y
 CONFIG_REGMAP=y
 CONFIG_SYSCON=y
 CONFIG_DEVRES=y
+CONFIG_ADC_MTK=y
 CONFIG_CLK=y
 CONFIG_DFU_MMC=y
 CONFIG_DFU_UFS=y
@@ -71,6 +75,7 @@ CONFIG_DM_I2C=y
 CONFIG_SYS_I2C_MTK=y
 # CONFIG_INPUT is not set
 # CONFIG_MMC_QUIRKS is not set
+CONFIG_MMC_MTK=y
 CONFIG_PHY_REALTEK=y
 CONFIG_PHY_ETHERNET_ID=y
 CONFIG_DWC_ETH_QOS=y
@@ -80,6 +85,12 @@ CONFIG_PHY_MTK_TPHY=y
 CONFIG_PINCTRL=y
 CONFIG_PINCONF=y
 CONFIG_PINCTRL_MT8195=y
+CONFIG_POWER_DOMAIN=y
+CONFIG_MTK_POWER_DOMAIN=y
+CONFIG_DM_PMIC=y
+CONFIG_MTK_PMIC_WRAP=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_MT6357=y
 CONFIG_DM_RTC=y
 CONFIG_RTC_EMULATION=y
 CONFIG_SCSI=y
-- 
Armbian

